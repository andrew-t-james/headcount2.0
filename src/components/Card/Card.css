.card {
  border: 1px solid red;
  border-radius: .4rem;
  box-shadow: 8px 10px 20px 0 rgba(46,61,73,.15);
  padding: 1rem;
  text-align: center;
}

.card-list {
  list-style: none;
}
