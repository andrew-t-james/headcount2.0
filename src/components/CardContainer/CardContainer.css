.card-container {
  display: grid;
  grid-column-gap: 15px;
  grid-template-columns: repeat(auto-fit, minmax(30rem, 1fr));
  margin-top: 5rem;
}