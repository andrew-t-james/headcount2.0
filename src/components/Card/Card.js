import React from 'react';
import PropTypes from 'prop-types';

import './Card.css';

const Card = ({ location, stats, selected, handleSelectedDistrict }) => {
  const districtStats = Object.keys(stats).map(stat =>
    <li className="card-stats" key={stat}>
      <span className="card-stats-year">{stat}</span>
      <span className={stats[stat] < 0.5 ? 'card-stats-low' : 'card-stats-high'}>
        {stats[stat]}
      </span>
    </li>
  );

  return (
    <div onClick={() => handleSelectedDistrict(location)} className={`card ${selected ? 'selected' : ''}`}>
      <h2 className="card-location">{location}</h2>
      <ul className={selected ? 'card-list' : 'card-list collapsed-list'}>
        {districtStats}
      </ul>
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired,
  selected: PropTypes.bool,
  handleSelectedDistrict: PropTypes.func
};

export default Card;