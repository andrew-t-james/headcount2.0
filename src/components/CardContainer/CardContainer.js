import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

import './CardContainer.css';


const CardContainer = ({ districts, handleSelectedDistrict}) => {
  const districtData = districts.map(district => {
    return (
      <Card
        {...district}
        key={district.id}
        handleSelectedDistrict={handleSelectedDistrict}
      />
    );
  });

  return (
    <div>
      { districtData &&
        <div className="card-container">
          {districtData}
        </div>
      }
      {
        !districtData.length &&
        <div className="card-container">
          <h2 className="card-container-error-message">Oops No District by that name</h2>
        </div>
      }
    </div>
  );
};

CardContainer.propTypes = {
  districts: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleSelectedDistrict: PropTypes.func
};

export default CardContainer;