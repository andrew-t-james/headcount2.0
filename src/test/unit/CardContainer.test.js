import React from 'react';
import CardContainer from '../../components/CardContainer/CardContainer';
import Card from '../../components/Card/Card';
import { shallow } from 'enzyme';

describe('CardContainer test suite', () => {
  let wrapper;
  const mockData = [
    {
      location:'The Moon',
      stats: {
        2002: 1,
        2003: 6,
        2004: 8,
        2005: 0,
      }
    },
    {
      location:'The Sun',
      stats: {
        2002: 1,
        2007: 6,
        2004: 8,
        2009: 0,
      }
    },
    {
      location:'Jupiter',
      stats: {
        2002: 1,
        2009: 6,
        2010: 8,
        2004: 0,
      }
    }
  ];

  beforeEach(() => wrapper = shallow(<CardContainer districts={mockData}/>));

  afterEach(() => wrapper.unmount());

  test('There should be three cards created in the CardContainer when given props array', () => {
    const cards = wrapper.find(Card).length;
    expect(cards).toBe(3);
  });

  test('When CardContainer is rendered it should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
