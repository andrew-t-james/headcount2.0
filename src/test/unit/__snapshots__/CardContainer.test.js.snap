// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer test suite When CardContainer is rendered it should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districts={
        Array [
            Object {
              "location": "The Moon",
              "stats": Object {
                "2002": 1,
                "2003": 6,
                "2004": 8,
                "2005": 0,
              },
            },
            Object {
              "location": "The Sun",
              "stats": Object {
                "2002": 1,
                "2004": 8,
                "2007": 6,
                "2009": 0,
              },
            },
            Object {
              "location": "Jupiter",
              "stats": Object {
                "2002": 1,
                "2004": 0,
                "2009": 6,
                "2010": 8,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Card Container
</h1>,
        <div
          className="card-container"
>
          <Card
                    location="The Moon"
                    stats={
                              Object {
                                        "2002": 1,
                                        "2003": 6,
                                        "2004": 8,
                                        "2005": 0,
                                      }
                    }
          />
          <Card
                    location="The Sun"
                    stats={
                              Object {
                                        "2002": 1,
                                        "2004": 8,
                                        "2007": 6,
                                        "2009": 0,
                                      }
                    }
          />
          <Card
                    location="Jupiter"
                    stats={
                              Object {
                                        "2002": 1,
                                        "2004": 0,
                                        "2009": 6,
                                        "2010": 8,
                                      }
                    }
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Card Container",
        },
        "ref": null,
        "rendered": "Card Container",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              location="The Moon"
              stats={
                            Object {
                                          "2002": 1,
                                          "2003": 6,
                                          "2004": 8,
                                          "2005": 0,
                                        }
              }
/>,
            <Card
              location="The Sun"
              stats={
                            Object {
                                          "2002": 1,
                                          "2004": 8,
                                          "2007": 6,
                                          "2009": 0,
                                        }
              }
/>,
            <Card
              location="Jupiter"
              stats={
                            Object {
                                          "2002": 1,
                                          "2004": 0,
                                          "2009": 6,
                                          "2010": 8,
                                        }
              }
/>,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "The Moon",
            "nodeType": "function",
            "props": Object {
              "location": "The Moon",
              "stats": Object {
                "2002": 1,
                "2003": 6,
                "2004": 8,
                "2005": 0,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "The Sun",
            "nodeType": "function",
            "props": Object {
              "location": "The Sun",
              "stats": Object {
                "2002": 1,
                "2004": 8,
                "2007": 6,
                "2009": 0,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Jupiter",
            "nodeType": "function",
            "props": Object {
              "location": "Jupiter",
              "stats": Object {
                "2002": 1,
                "2004": 0,
                "2009": 6,
                "2010": 8,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Card Container
</h1>,
          <div
            className="card-container"
>
            <Card
                        location="The Moon"
                        stats={
                                    Object {
                                                "2002": 1,
                                                "2003": 6,
                                                "2004": 8,
                                                "2005": 0,
                                              }
                        }
            />
            <Card
                        location="The Sun"
                        stats={
                                    Object {
                                                "2002": 1,
                                                "2004": 8,
                                                "2007": 6,
                                                "2009": 0,
                                              }
                        }
            />
            <Card
                        location="Jupiter"
                        stats={
                                    Object {
                                                "2002": 1,
                                                "2004": 0,
                                                "2009": 6,
                                                "2010": 8,
                                              }
                        }
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Card Container",
          },
          "ref": null,
          "rendered": "Card Container",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                location="The Moon"
                stats={
                                Object {
                                                "2002": 1,
                                                "2003": 6,
                                                "2004": 8,
                                                "2005": 0,
                                              }
                }
/>,
              <Card
                location="The Sun"
                stats={
                                Object {
                                                "2002": 1,
                                                "2004": 8,
                                                "2007": 6,
                                                "2009": 0,
                                              }
                }
/>,
              <Card
                location="Jupiter"
                stats={
                                Object {
                                                "2002": 1,
                                                "2004": 0,
                                                "2009": 6,
                                                "2010": 8,
                                              }
                }
/>,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "The Moon",
              "nodeType": "function",
              "props": Object {
                "location": "The Moon",
                "stats": Object {
                  "2002": 1,
                  "2003": 6,
                  "2004": 8,
                  "2005": 0,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "The Sun",
              "nodeType": "function",
              "props": Object {
                "location": "The Sun",
                "stats": Object {
                  "2002": 1,
                  "2004": 8,
                  "2007": 6,
                  "2009": 0,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Jupiter",
              "nodeType": "function",
              "props": Object {
                "location": "Jupiter",
                "stats": Object {
                  "2002": 1,
                  "2004": 0,
                  "2009": 6,
                  "2010": 8,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
